### PR TITLE
ENH: Synchronize XML

### PR DESCRIPTION
**Pull request by bashtage**
Synchronize to and read XML to upstream

### CODE REVIEW

**Comment by bashtage** [comment]
No tests because lxml is not in the environment. Not sure if there is a desire to expand the install.

File: pandas-stubs/core/frame.pyi [hunk 1]
```diff
@@ -328,6 +333,46 @@ class DataFrame(NDFrame, OpsMixin):
         render_links: _bool = ...,
         encoding: _str | None = ...,
     ) -> _str: ...
+    @overload
+    def to_xml(
+        self,
+        path_or_buffer: FilePath | WriteBuffer[bytes] | WriteBuffer[str],
+        index: bool = ...,
+        root_name: str | None = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/core/frame.pyi:341]: Docs don't show `None` as option for `root_name`

File: pandas-stubs/core/frame.pyi [hunk 2]
```diff
@@ -328,6 +333,46 @@ class DataFrame(NDFrame, OpsMixin):
...
+        row_name: str | None = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/core/frame.pyi:342]: Docs don't show `None` as option for `row_name`

File: pandas-stubs/core/frame.pyi [hunk 3]
```diff
@@ -328,6 +333,46 @@ class DataFrame(NDFrame, OpsMixin):
...
+        na_rep: str | None = ...,
+        attr_cols: list[HashableT] | None = ...,
+        elem_cols: list[HashableT] | None = ...,
+        namespaces: dict[str | None, str] | None = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/core/frame.pyi:346]: Is `None` really allowed as a valid key in the `dict` for `namespaces` ?

File: pandas-stubs/core/frame.pyi [hunk 4]
```diff
@@ -328,6 +333,46 @@ class DataFrame(NDFrame, OpsMixin):
...
+        prefix: str | None = ...,
+        encoding: str = ...,
+        xml_declaration: bool | None = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/core/frame.pyi:349]: Docs don't show `None` as option for `xml_declaration`

File: pandas-stubs/core/frame.pyi [hunk 5]
```diff
@@ -328,6 +333,46 @@ class DataFrame(NDFrame, OpsMixin):
...
+        pretty_print: bool | None = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/core/frame.pyi:350]: Docs don't show `None` as option for `pretty_print`

File: pandas-stubs/core/frame.pyi [hunk 6]
```diff
@@ -328,6 +333,46 @@ class DataFrame(NDFrame, OpsMixin):
...
+        parser: str | None = ...,
```
Comment (Dr-Irv) [inline pandas-stubs/core/frame.pyi:351]: docs show `Literal["lxml", "etree"]` for `parser`

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

**Comment by Dr-Irv** [comment]
> No tests because lxml is not in the environment. Not sure if there is a desire to expand the install.

It would just have to go in the section in `pyproject.toml` under `[tool.poetry.dev-dependencies]` so wouldn't be an installation requirement for the stubs, just for testing, and I think we should do that.

File: pandas-stubs/core/frame.pyi (hunk 3 above)
Comment (bashtage) [reply pandas-stubs/core/frame.pyi:354]: `None` works in code.

File: pandas-stubs/core/frame.pyi (hunk 4 above)
Comment (bashtage) [reply pandas-stubs/core/frame.pyi:357]: It is the default value, despite claiming False.

File: pandas-stubs/core/frame.pyi (hunk 2 above)
Comment (bashtage) [reply pandas-stubs/core/frame.pyi:350]: Fixed.

File: pandas-stubs/core/frame.pyi (hunk 1 above)
Comment (bashtage) [reply pandas-stubs/core/frame.pyi:349]: Fixed.

File: pandas-stubs/core/frame.pyi (hunk 5 above)
Comment (bashtage) [reply pandas-stubs/core/frame.pyi:358]: Fixed.

File: pandas-stubs/core/frame.pyi (hunk 4 above)
Comment (bashtage) [reply pandas-stubs/core/frame.pyi]: Fixed.

**Review by Dr-Irv — APPROVED** [review]
thanks @bashtage

**Pull request merged**